subsys/usb/usb_device.c

```diff
@@ -1229,7 +1229,8 @@ static int class_handler(struct usb_setup_packet *pSetup,
 		}
 
 		if (iface->class_handler &&
-		    if_descr->bInterfaceNumber == (pSetup->wIndex & 0xFF)) {
+		    (if_descr->bInterfaceNumber == ((pSetup->wIndex) & 0xFF) ||
```

Comment:
This is an exception for audio Requests with bmRequestType = `00100010b` (class request type) and `pSetup->wIndex` is endpoint number in this case (not interface number as it was working until now). This is another issue with core<->class interface we have. We cannot follow those exceptions with forwarding requests. We need to rethink validity of specifying 'class/vendor/custom' handlers at all.

Reply:
Sorry. `class/vendor/custom` should stay but the way we are forwarding is incorrect. This I will address during the rework.

Reply:
Yes, but the key is "once the information about interfaces is available for the core" :p Right now its not, but we are heading this.

Reply:
okay, but I still do not see how commit message reflects the changes and why.

Reply:
Sorry about that, the commit message was incorrect. I did the change so that class endpoint requests would be forwarded to the class handler without matching interface number, just like it is done for the custom handler.
It forwards all class requests to the audio class handler so this change might break the usage of CDC and AUDIO classes in the same application. Hopefully the rewritten usb core will address this issue?

Reply:
Thanks. The issue will be addressed.